### PR TITLE
Fixes #18141, eliminating null in value when iterating over `ENV`.

### LINKED ISSUE #18141: Iterator over `ENV` leaves extra null terminator in values on windows.
A (truncated) dump of `ENV`:
```
julia> ENV
Base.EnvHash with 71 entries:
  "=::"                      => "::\\\0"
  "ALLUSERSPROFILE"          => "C:\\ProgramData\0"
  "APPDATA"                  => "C:\\Users\\hw9980\\AppData\\Roaming\0"
  "COMMONPROGRAMFILES"       => "C:\\Program Files\\Common Files\0"
  "COMPUTERNAME"             => "SNC39466\0"
  "COMSPEC"                  => "C:\\WINDOWS\\system32\\cmd.exe\0"
  "ChocolateyInstall"        => "C:\\ProgramData\\chocolatey\0"
  "ChocolateyLastPathUpdate" => "Tue Jul 19 11:05:47 2016\0"
  "CommonProgramFiles(x86)"  => "C:\\Program Files (x86)\\Common Files\0"
  "CommonProgramW6432"       => "C:\\Program Files\\Common Files\0"
```

### PR DESCRIPTION
**Pull request by twadleigh**
Fixes #18141.

### CODE REVIEW

**Comment by tkelman** [comment]
Could you add a test that would have failed without this fix?

**Comment by twadleigh** [comment]
@tkelman, will do.

File: test/env.jl [hunk 1]
```diff
@@ -63,3 +63,8 @@ end
 
 # Test for #10853
 @test withenv(Dict{Any,Any}()...) do; true; end
+
+# Test for #18141
+for (k, v) in ENV
+    @test v[end] != '\0'
```
Comment (tkelman) [inline test/env.jl]: I wonder whether it's possible to have this happen from outside of Julia?

Comment (twadleigh) [inline test/env.jl]: I don't follow.

Comment (tkelman) [inline test/env.jl]: Could anyone have this be the case in their pre-existing environment, or would our getenv always be reading based on null-termination?

**Pull request merged**